src/load_inputs/load_dataframe.jl

```diff
@@ -1,19 +1,83 @@
+function filenotfoundconstant()
+    "FILENOTFOUND"
+end
+
+@doc raw"""
+    file_exists(dir::AbstractString, basenames::Vector{String})::Bool
+
+Checks that a file exists in a directory under (at least) one of a list of 'aliases'.
+"""
+function file_exists(dir, basenames::Vector{String})::Bool
+    if !isdir(dir)
+        return false
+    end
+    best_basename = popfirst!(basenames)
+    best_path = joinpath(dir, best_basename)
+    if isfile(best_path)
+        return true
+    end
+
+    FILENOTFOUND = filenotfoundconstant()
+
+    for base in basenames
+        target = look_for_file_with_alternate_case(dir, base)
+        if target != FILENOTFOUND
+            return true
+        end
+    end
+    false
+end
+
 @doc raw"""
     load_dataframe(path::AbstractString)
 
 Attempts to load a dataframe from a csv file with the given path.
 If it's not found immediately, it will look for files with a different case (lower/upper)
 in the file's basename.
 """
-function load_dataframe(path::AbstractString)
-    if isfile(path)
-        return load_dataframe_from_file(path)
+function load_dataframe(path::AbstractString)::DataFrame
+    dir, base = dirname(path), basename(path)
+    load_dataframe(dir, [base])
+end
+
+@doc raw"""
+    load_dataframe(dir::AbstractString, base::AbstractString)
+
+Attempts to load a dataframe from a csv file with the given directory and file name.
+If not found immediately, look for files with a different case (lower/upper)
+in the file's basename.
+"""
+function load_dataframe(dir::AbstractString, base::AbstractString)::DataFrame
+    load_dataframe(dir, [base])
+end
+
+function load_dataframe(dir::AbstractString, basenames::Vector{String})::DataFrame
+    best_basename = popfirst!(basenames)
+    best_path = joinpath(dir, best_basename)
+    if isfile(best_path)
+        return load_dataframe_from_file(best_path)
     end
 
-    # not immediately found
-    dir, base = dirname(path), basename(path)
-    target = look_for_file_with_alternate_case(dir, base)
-    load_dataframe_from_file(joinpath(dir, target))
+    FILENOTFOUND = filenotfoundconstant()
+
+    for base in basenames
+        target = look_for_file_with_alternate_case(dir, base)
+        # admonish
+        if target != FILENOTFOUND
+            @info """The filename '$target' is deprecated. '$best_basename' is preferred."""
+            return load_dataframe_from_file(joinpath(dir, target))
+        end
+    end
+
+    throw_filenotfound_error(dir, best_basename)
+end
+
+function throw_filenotfound_error(dir, base)
+    files_in_dir = readdir(dir)
+    err_str = """File $base was not found in the directory, "$dir".
+                 Try checking the spelling.
+                 The files in the directory are $files_in_dir."""
+    error(err_str)
 end
```

Comment:
Arguably one should never `error`, only throw things, in case someone using GenX as a package wants to catch these.